dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java

```diff
@@ -404,6 +404,13 @@ protected void saveLanguage(final Language lang) {
 			}
 			dbUpsert(lang);
 			CacheLocator.getLanguageCache().clearLanguages();
+			// Cleaning up the createDefaultLanguageLock and the default language cache record
+			// if we are updating the default language, otherwise changes won't be reflected.
+			if (createDefaultLanguageLock.get() != null
```

Comment:
Shouldn't this be done within a `createDefaultLanguageLock.compareAndSet` operation?
Like :
if(createDefaultLanguageLock.compareAndSet(lang, defaultLanguage){
  //clear the cache & update the lock 
}

Reply:
The reason I would leave where it is now, is because we should not be cleaning cache on reads, the `saveLanguage` it is already cleaning everything it needs to be removed from cache, I would need to add the cleanup in the `getDefaultLanguage()`, after the `if (createDefaultLanguageLock.compareAndSet(null, defaultLanguage))` and not sure if it is the best place.

Reply:
My suggestion was more around executing these two operations within a compareAndSet 

```
CacheLocator.getLanguageCache().clearDefaultLanguage();
				createDefaultLanguageLock.set(null);
```
But doing it in the save makes perfect  sense